mali: support building against 4.14

In 4.14, global_page_state is renamed to global_zone_page_state

See commit c41f012ade0b ("mm: rename global_page_state to
global_zone_page_state")

Signed-off-by: Jonathan Liu <net147@gmail.com>
--- a/src/devicedrv/mali/linux/mali_memory_swap_alloc.c
+++ b/src/devicedrv/mali/linux/mali_memory_swap_alloc.c
@@ -257,7 +257,11 @@ static void mali_mem_swap_swapped_bkend_pool_shrink(_mali_mem_swap_pool_shrink_t
 	}
 
 	/* Get system free pages number. */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 14, 0)
+	system_free_size = global_zone_page_state(NR_FREE_PAGES) * PAGE_SIZE;
+#else
 	system_free_size = global_page_state(NR_FREE_PAGES) * PAGE_SIZE;
+#endif
 	last_gpu_utilization = _mali_ukk_utilization_gp_pp();
 
 	if ((last_gpu_utilization < gpu_utilization_threshold_value)
@@ -585,7 +589,11 @@ int mali_mem_swap_alloc_pages(mali_mem_swap *swap_mem, u32 size, u32 *bkend_idx)
 		list_add_tail(&m_page->list, &swap_mem->pages);
 	}
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 14, 0)
+	system_free_size = global_zone_page_state(NR_FREE_PAGES) * PAGE_SIZE;
+#else
 	system_free_size = global_page_state(NR_FREE_PAGES) * PAGE_SIZE;
+#endif
 
 	if ((system_free_size < mali_mem_swap_out_threshold_value)
 	    && (mem_backend_swapped_pool_size > (mali_mem_swap_out_threshold_value >> 2))
